### PR TITLE
Fix: text generation error when no bonds are detected at site in cation-anion analysis

### PR DESCRIPTION
**Pull request by naik-aakash**
Addresses the atomate2 post processing issue

https://github.com/materialsproject/atomate2/issues/681

### CODE REVIEW

File: lobsterpy/cohp/analyze.py [hunk 1]
```diff
@@ -236,14 +236,21 @@ def __init__(self, chemenv, valences=None):
 
                     """
                     if valences is None:
-                        self.coordination_environments = [
-                            [{"ce_symbol": str(len(coord))}] for coord in chemenv
-                        ]
+                        self.coordination_environments = []
+                        for coord in chemenv:
+                            if coord != []:
```
Comment (JaGeo) [inline lobsterpy/cohp/analyze.py:241]: You should take the length instead. This is notnsafe.

Comment (naik-aakash) [reply lobsterpy/cohp/analyze.py:241]: Fixed it

File: lobsterpy/cohp/analyze.py [hunk 2]
```diff
@@ -236,14 +236,21 @@ def __init__(self, chemenv, valences=None):
...
+                            else:
+                                self.coordination_environments.append(
+                                    [{"ce_symbol": None}]
+                                )
                     else:
                         self.coordination_environments = []
 
                         for val, coord in zip(valences, chemenv):
-                            if val >= 0.0:
+                            if val >= 0.0 and coord != []:
```
[the remark this answers is not in the document]
Comment (naik-aakash) [reply lobsterpy/cohp/analyze.py:253]: Fixed it

**Review by JaGeo — APPROVED** [review]

**Comment by JaGeo** [comment]
Thanks, @naik-aakash .

I will release a new version, then you can fix atomate2 and look into the JOSS submission process!

**Pull request merged**